### ISSUE TITLE
vmm: Wait for completion of PCI device removal

### ISSUE THREAD

**Issue by sboeuf** [cloud-hypervisor/cloud-hypervisor#4038]
A race condition can happen if trying to unplug a device and plug a new device with the same identifier right after. This is due to the behavior of the `remove-device` request, as it's not waiting for the removal to be fully complete.
We must introduce a synchronization mechanism to avoid this behavior and prevent that kind of issue from happening.

Based on the feedback from @fidencio [here](https://github.com/cloud-hypervisor/cloud-hypervisor/pull/4025#issuecomment-1114533019)

**Comment by rbradford** [comment]
Sounds like you need to check identifiers in both the config and the `DeviceTree` to check that the same name isn't being used in a device that is removed from the config but that is still present and active because the guest hasn't yet done an eject or never will.

**Comment by sboeuf** [comment]
> Sounds like you need to check identifiers in both the config and the `DeviceTree` to check that the same name isn't being used in a device that is removed from the config but that is still present and active because the guest hasn't yet done an eject or never will.

Yes I agree we can reinforce the check for the device identifier being unique, which will result in a more appropriate error for Kata Containers. But in the end, it's only with the `event-monitor` that this can be solved.
